File: app/src/main/java/co/basic/androidjetpackskeleton/adapters/RecyclerViewAdapter.java
package co.basic.androidjetpackskeleton.adapters;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import co.basic.androidjetpackskeleton.R;
import co.basic.androidjetpackskeleton.model.Data;
import co.basic.androidjetpackskeleton.networking.ApiClient;
import co.basic.androidjetpackskeleton.networking.ApiInterface;


public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {


    private Context context;
    private ArrayList<Data> dataList;
    private LayoutInflater mInflater;

    // data is passed into the constructor
    public RecyclerViewAdapter(Context context, ArrayList<Data> dataList) {
        this.context = context;
        this.mInflater = LayoutInflater.from(context);
        this.dataList = dataList;
    }

    // inflates the row layout from xml when needed
    @Override
    @NonNull
    public RecyclerViewAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.raw_recyclerview_item, parent, false);
        return new RecyclerViewAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder viewHolder, int position) {


        Log.d("image_url",dataList.get(viewHolder.getAdapterPosition()).getPosterPath());


        String image = dataList.get(position).getPosterPath().substring(1);
        final int id = dataList.get(position).getId();


        Picasso.get().load(ApiClient.IMAGE_URL+image).into(viewHolder.ivListItem);


    }


    // total number of rows
    @Override
    public int getItemCount() {
        return dataList.size();
    }



    // stores and recycles views as they are scrolled off screen
    static class ViewHolder extends RecyclerView.ViewHolder {

        ImageView ivListItem;


        ViewHolder(View itemView) {
            super(itemView);

            ivListItem = itemView.findViewById(R.id.iv_list_item);

        }
    }
}

File: app/src/main/java/co/basic/androidjetpackskeleton/adapters/RecyclerViewAdapter.kt
package co.basic.androidjetpackskeleton.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import co.basic.androidjetpackskeleton.R
import co.basic.androidjetpackskeleton.model.Data
import co.basic.androidjetpackskeleton.networking.ApiClient
import co.basic.androidjetpackskeleton.OnItemClickListener
import com.squareup.picasso.Picasso
import java.util.*

class RecyclerViewAdapter(
    private val context: Context,
    dataList: ArrayList<Data>,
    private val listener: OnItemClickListener
) :

    RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {

    interface ClickListener {

        fun onClickListener(item: ArrayList<Data>)
    }


    private val dataList: ArrayList<Data>
    private val mInflater: LayoutInflater

    var id: Int = 1


    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = mInflater.inflate(R.layout.raw_recyclerview_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val image = dataList[position].posterPath.substring(1)
        id = dataList[position].id


        Picasso.get().load(ApiClient.IMAGE_URL + image).into(viewHolder.ivListItem)

        viewHolder.itemView.setOnClickListener {

            listener.onItemClick(
                viewHolder.adapterPosition,
                dataList.get(viewHolder.adapterPosition)
            )


        }


    }

    // total number of rows
    override fun getItemCount(): Int {
        return dataList.size
    }

    // stores and recycles views as they are scrolled off screen
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var ivListItem: ImageView

        init {
            ivListItem = itemView.findViewById(R.id.iv_list_item)

        }
    }

    // data is passed into the constructor
    init {
        mInflater = LayoutInflater.from(context)
        this.dataList = dataList
    }
}

